kernel/arch/dreamcast/sound/snd_stream.c

```diff
@@ -151,16 +151,40 @@ static void process_filters(snd_stream_hnd_t hnd, void **buffer, int *samplecnt)
 }
 
 static void sep_data(void *buffer, int len, int stereo) {
-    uint32_t data;
     uint32_t *buf = (uint32_t *)buffer;
-    uint16_t *left = (uint16_t *)sep_buffer[1];
-    uint16_t *right = (uint16_t *)sep_buffer[0];
+    uint32_t *left_ptr = (uint32_t *)sep_buffer[1];
+    uint32_t *right_ptr = (uint32_t *)sep_buffer[0];
+    uint32_t data;
+    uint32_t left_val;
+    uint32_t right_val;
 
     if(stereo) {
-        for(; len > 0; len -= 2) {
+        len <<= 1;
+        for(; len > 8; len -= 8) {
+            if(((uintptr_t)buf & 31) == 0) {
+                dcache_pref_block(buf + 8);
+            }
+            data = *buf++;
+            left_val = (data >> 16);
+            right_val = (data & 0xffff);
+
+            data = *buf++;
+            left_val |= (data & 0xffff0000);
+            right_val |= (data & 0xffff) << 16;
+
+            if(((uintptr_t)left_ptr & 31) == 0) {
```

Comment:
Same here, I believe the `if` can be removed, and calling `dcache_alloc_block` unconditionally should work fine.

Reply:
No here it's not works, sound with distortions.